File: Home2WorkAPI/src/main/java/it/gruppoinfor/home2workapi/model/UserProfile.java
package it.gruppoinfor.home2workapi.model;


import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.Date;
import java.util.List;

public class UserProfile implements Serializable {

    @SerializedName("Regdate")
    @Expose
    private Date registrationDate;

    @SerializedName("Exp")
    @Expose
    private Experience exp;
    @SerializedName("Stats")
    @Expose
    private Statistics stats;
    @SerializedName("Achievements")
    @Expose
    private List<Achievement> achievements;

    public Date getRegistrationDate() {
        return registrationDate;
    }

    public void setRegistrationDate(Date registrationDate) {
        this.registrationDate = registrationDate;
    }

    public Experience getExp() {
        return exp;
    }

    public void setExp(Experience exp) {
        this.exp = exp;
    }

    public Statistics getStats() {
        return stats;
    }

    public void setStats(Statistics stats) {
        this.stats = stats;
    }

    public List<Achievement> getAchievements() {
        return achievements;
    }

    public void setAchievements(List<Achievement> achievements) {
        this.achievements = achievements;
    }
}

File: Home2WorkAPI/src/main/java/it/gruppoinfor/home2workapi/model/UserProfile.kt
package it.gruppoinfor.home2workapi.model


import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.io.Serializable
import java.util.*
import kotlin.collections.ArrayList

class UserProfile : Serializable {

    @SerializedName("Regdate")
    @Expose
    var registrationDate: Date? = null
    @SerializedName("Exp")
    @Expose
    var exp: Experience = Experience()
    @SerializedName("Stats")
    @Expose
    var stats: Statistics = Statistics()
    @SerializedName("Activity")
    @Expose
    var activity: List<MonthActivity> = ArrayList<MonthActivity>()
}
